Img src cdn

Rajatuba-Cogoport opened the pull request.

Rajatuba-Cogoport committed:
img-src-cdn

Rajatuba-Cogoport committed:
removed console

Rajatuba-Cogoport committed:
version up

Rajatuba-Cogoport committed:
messgae changed and ignore global constant file

Rajatuba-Cogoport committed:
regex pattern check

Rajatuba-Cogoport committed:
omit geo file

Rajatuba-Cogoport committed:
nbsp-check

Rajatuba-Cogoport committed:
ignore global and new url for img src added

Rajatuba-Cogoport committed:
message changes

Rajatuba-Cogoport committed:
is-empty-use-check

Rajatuba-Cogoport committed:
Merge branch 'main' into img_src_cdn

Rajatuba-Cogoport committed:
move into package

Rajatuba-Cogoport committed:
change

Rajatuba-Cogoport committed:
change

Rajatuba-Cogoport committed:
move ignore to custom-rules

Rajatuba-Cogoport committed:
warn for cdn check

Rajatuba-Cogoport committed:
changes

Rajatuba-Cogoport committed:
space

Rajatuba-Cogoport committed:
changes

Rajatuba-Cogoport committed:
extra space

Rajatuba-Cogoport committed:
moved in package

Rajatuba-Cogoport committed:
Merge pull request #8 from Cogoport/is-empty-check

is-empty-use-check

Rajatuba-Cogoport committed:
Merge pull request #7 from Cogoport/nbsp-check

nbsp-check

Rajatuba-Cogoport committed:
Merge pull request #6 from Cogoport/regex-check

Regex check

Rajatuba-Cogoport committed:
removed custom_rules

Rajatuba-Cogoport committed:
remove git action

Rajatuba-Cogoport committed:
beta version

prashantp01 committed:
🔖 chore(package.json): update package version from 1.1.1-beta.1 to 1.1.1-beta.10

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

Rajatuba-Cogoport committed:
fix

Rajatuba-Cogoport committed:
uuid_check

Rajatuba-Cogoport committed:
final

Rajatuba-Cogoport committed:
message change

Rajatuba-Cogoport committed:
email check

Rajatuba-Cogoport committed:
Merge pull request #9 from Cogoport/uuid_and_email

UUID and email

Rajatuba-Cogoport committed:
cl use

Rajatuba-Cogoport committed:
Merge pull request #10 from Cogoport/js_to_json

cl use

Rajatuba-Cogoport committed:
date time format check

Rajatuba-Cogoport committed:
Merge pull request #11 from Cogoport/date_format

date time format check

Rajatuba-Cogoport committed:
role button

Rajatuba-Cogoport committed:
Merge pull request #12 from Cogoport/div_role

role button

Roh-Patil committed:
fix: adding zeroth index rule

Rajatuba-Cogoport committed:
key checks

Roh-Patil committed:
Merge branch 'img_src_cdn' of github.com:Cogoport/eslint-config into zeroth-index-import

Roh-Patil committed:
zeroth index: adding imports

Rajatuba-Cogoport committed:
Merge pull request #13 from Cogoport/zeroth-index-import

Zeroth-index-import

Rajatuba-Cogoport committed:
merge conf

Rajatuba-Cogoport committed:
Merge pull request #14 from Cogoport/uuid_check

key checks

Commit:
adding pascalcase rule

Commit:
Revert adding pascalcase rule

Shweta-819 committed:
fix: rule

Rajatuba-Cogoport committed:
Merge pull request #20 from Cogoport/pascal_new_component

fix: rule

Roh-Patil committed:
refactor

Rajatuba-Cogoport committed:
hook sequence

Rajatuba-Cogoport committed:
conf

Rajatuba-Cogoport committed:
Merge pull request #22 from Cogoport/uuid_check

hook sequence

Shweta-819 committed:
fix: usestate

Shweta-819 committed:
Merge branch 'img_src_cdn' of github.com:Cogoport/eslint-config into usestate_rule

Roh-Patil committed:
variable name check within functional component

Roh-Patil committed:
Merge branch 'img_src_cdn' of github.com:Cogoport/eslint-config into variable_camelCase

Rajatuba-Cogoport committed:
changes

Rajatuba-Cogoport committed:
Merge pull request #25 from Cogoport/uuid_check

changes

Rajatuba-Cogoport committed:
changes

Rajatuba-Cogoport committed:
Merge pull request #26 from Cogoport/uuid_check

changes

Shweta-819 committed:
fix: usestate

Shweta-819 committed:
fix: usestate

Roh-Patil committed:
constant zero value declaration handled

Roh-Patil committed:
code refactor

Shweta-819 committed:
fix: rule

Rajatuba-Cogoport committed:
Merge pull request #24 from Cogoport/usestate_rule

fix: usestate

Roh-Patil committed:
variable check condition

Roh-Patil committed:
adding optional chaining

Roh-Patil committed:
Merge branch 'img_src_cdn' of github.com:Cogoport/eslint-config into variable_camelCase

Rajatuba-Cogoport committed:
props default

Roh-Patil committed:
Merge branch 'img_src_cdn' into variable_camelCase

Rajatuba-Cogoport committed:
Merge pull request #21 from Cogoport/variable_camelCase

snake_case & camelCase variable

Roh-Patil committed:
multiple zero-value constants handled

Rajatuba-Cogoport committed:
Merge pull request #27 from Cogoport/zeroth-index-fix

Zeroth-index-fix

Rajatuba-Cogoport committed:
null handling

Rajatuba-Cogoport committed:
value can be jsx

Rajatuba-Cogoport committed:
pnpm update

prashantp01 commented on custom-rules.js:
```diff
@@ -2,14 +2,33 @@ const overrides = require('./overrides');
+	extends        : ['airbnb', 'airbnb/hooks'],
+	plugins        : ['custom-rules'],
+	parserOptions  : { ecmaVersion: 2020 },
+	ignorePatterns : ['**/globalization/*.js'],
```
remove ignore patherns

Rajatuba-Cogoport committed:
custom-rules -> custom-lint

Rajatuba-Cogoport committed:
changes

Rajatuba-Cogoport committed:
eslint-plugin-custom-rules in package.json

prashantp01 committed:
📦 chore(custom-eslint-rules.js): add custom ESLint rules for project linting

✨ feat(check_element_role_button.js): add custom ESLint rule to check if element is converted to button

✨ feat(cl_for_merging_styles_class.js): add custom ESLint rule to check for the use of cl to merge styles class

✨ feat(component_pascal.js): add custom ESLint rule to enforce PascalCase for React component names and props default

✨ feat(custom_rules_matching.js): add custom ESLint rule to check variable naming conventions

✨ feat(date_time_format_check.js): add custom ESLint rule to check if dateFormat is not from GLOBAL_CONSTANTS.formats.dd/MM/yyyy

✨ feat(default_comp_props.js): add custom ESLint rule to check for default values in React component props

✨ feat(default_usestate.js): add custom ESLint rule to check if default value is provided for useState

✨ feat(email_check

📦 feat(custom-eslint): add new custom ESLint rules

✨ feat(custom-eslint/hook_sequence.js): add new ESLint rule to enforce hook sequence in React components

✨ feat(custom-eslint/img_src_cdn.js): add new ESLint rule to enforce importing image src from global constants if it includes a specific URL

✨ feat(custom-eslint/is_empty_use_check.js): add new ESLint rule to enforce the use of isEmpty when checking variable length

✨ feat(custom-eslint/key_as_function.js): add new ESLint rule to disallow function as key in JSX attributes

✨ feat(custom-eslint/key_in_for_each_map.js): add new ESLint rule to ensure that JSX elements pushed into an array have the "key" prop

📦 chore(custom-eslint): add custom ESLint rules for linting

✨ feat(custom-eslint): add nbsp_ensp_check rule to disallow the usage of non-breaking space or en space in JSXText

✨ feat(custom-eslint): add regex_check rule to disallow regular expression patterns in Literal nodes

✨ feat(custom-eslint): add uuid_check rule to check for the presence of UUID and emit a warning in Literal nodes

✨ feat(custom-eslint): add variable_name_check rule to enforce camelCase or snake_case naming for variables declared within functional components in VariableDeclarator nodes

✨ feat(custom-eslint): add variable_value_jsx rule to ensure that variable values are not JSX in VariableDeclarator nodes

✨ feat(custom-eslint): add zeroth_index_import rule to enforce importing the zeroth index from global constants if it is an array or string in MemberExpression nodes

📦 chore(custom-stylelint): add custom Stylelint

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔧 chore(custom-eslint): update rule names to follow the new naming convention
🔧 chore(package.json): update version of @cogoport/eslint-config to 1.1.1-beta.17
🔧 chore(custom-eslint): update package name to eslint-plugin-custom-eslint
🔧 chore(custom-stylelint): update package name to stylelint-plugin-custom-style

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔧 chore(custom-eslint): add new custom eslint rule 'check_element_role_button'
🔧 chore(custom-eslint): add new custom eslint rule 'cl_for_merging_styles_class'
🔧 chore(custom-eslint): add new custom eslint rule 'component_pascal'
🔧 chore(custom-eslint): add new custom eslint rule 'date_time_format_check'
🔧 chore(custom-eslint): add new custom eslint rule 'default_comp_props'
🔧 chore(custom-eslint): add new custom eslint rule 'default_usestate'
🔧 chore(custom-eslint): add new custom eslint rule 'email_check'
🔧 chore(custom-eslint): add new custom eslint rule 'hook_sequence'

🔧 chore(custom-eslint): add new custom ESLint rule 'img_src_cdn.js' to enforce moving image URLs to global constants
🔧 chore(custom-eslint): add new custom ESLint rule 'is_empty_use_check.js' to enforce using isEmpty when checking variable length
🔧 chore(custom-eslint): add new custom ESLint rule 'key_as_function.js' to disallow using functions as keys
🔧 chore(custom-eslint): add new custom ESLint rule 'key_in_for_each_map.js' to ensure JSX elements pushed into an array have the "key" prop
🔧 chore(custom-eslint): add new custom ESLint rule 'nbsp_ensp_check.js' to disallow the usage of non-breaking space or en space
🔧 chore(custom-eslint): add new custom ESLint rule 'regex_check.js' to disallow regular expression patterns
🔧 chore(custom-eslint): add new custom ESLint rule 'uuid_check

📦 chore(custom-eslint): add new custom eslint rule 'variable_value_jsx.js' to ensure variable value is not JSX
📦 chore(custom-eslint): add new custom eslint rule 'variables_name_check.js' to enforce variable naming conventions
📦 chore(custom-eslint): add new custom eslint rule 'zeroth_index_import.js' to enforce importing the zeroth index from global constants

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔀 refactor(custom-eslint): move custom eslint rules to a separate 'rules' directory for better organization and maintainability
🔀 refactor(custom-eslint): rename 'custom-eslint-matching' rule to 'variables-name-check' for better clarity and consistency

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔧 chore(custom-eslint-rules.js): update ESLint rule 'custom-eslint/custom-eslint-matching' to 'custom-eslint/variables-name-check' for improved clarity and consistency

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔄 chore(package.json): update package version from 1.1.1-beta.17 to 2.0.0 for a major release

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔥 chore(.gitignore): remove trailing whitespace and add missing newline at the end of the file

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔥 chore(.gitignore): remove unnecessary line for **/.history directory

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>

prashantp01 committed:
🔄 chore(CODEOWNERS): update code owners to include @prashantp01 in addition to @Cogoport/platform-engineering for all files in the repository

Signed-off-by: prashantp01 <prashant.paddune@cogoport.com>
